src/Core/Checkout/Order/OrderException.php

```diff
@@ -193,22 +200,36 @@ public static function orderCustomerDeleted(string $orderId): CustomerDeletedExc
         return new CustomerDeletedException($orderId);
     }
 
+    /**
+     * @deprecated tag:v6.8.0 - reason:return-type-change - Will return GuestNotAuthenticatedException
+     */
     public static function guestNotAuthenticated(): self
     {
-        return new self(
-            Response::HTTP_FORBIDDEN,
-            self::CHECKOUT_GUEST_NOT_AUTHENTICATED,
-            'Guest not authenticated.'
-        );
+        if (!Feature::isActive('v6.8.0.0')) {
+            return new self(
+                Response::HTTP_FORBIDDEN,
+                self::CHECKOUT_GUEST_NOT_AUTHENTICATED,
+                'Guest not authenticated.'
+            );
+        }
+
+        return new GuestNotAuthenticatedException();
```

Comment:
I don't think it makes sense to deprecate and re-deprecate behind a feature flag. We're just restoring the 6.6.x behaviour here. 6.7 hasn't been released yet, nor is it in the RC phase, we should do the "BC" again and revert the behaviour of 6.6.x. It's silly to cripple just because the deprecation was misplaced in the last major :thinking:

Reply:
I agree that we should just do the BC due to 6.7 not being RC yet. But I'd like to get @mstegmeyer and @mitelg input here as well.

Reply:
🤷‍♂️ i don't mind either way. hard to decide.

Reply:
yes, we can directly use the specific exception and do not do another deprecation, just because it was introduced in the last weeks.